src/main/java/com/rackspace/jenkins_nodepool/NodePoolSlave.java

```diff
@@ -108,11 +120,79 @@ private void readObject(java.io.ObjectInputStream in)
         Jenkins.getInstance().removeNode(this);
     }
 
-    // TODO seems odd storing the computer object on the node since it's related to the slave object here.
     @Override
     public Computer createComputer() {
         NodePoolComputer npc = new NodePoolComputer(this, nodePoolNode);
         return npc;
     }
 
-}
+    /**
+     * This gets invoked when the user clicks "Save" while editing the "Node" in the UI.
+     * @param req  stapler request
+     * @param form form data to update slave with
+     * @return new slave object
+     * @throws Descriptor.FormException  if things go sideways
+     */
+    @Override
+    public Node reconfigure(final StaplerRequest req, JSONObject form) throws Descriptor.FormException {
+        // update `this` with submitted data from the form and return it.  the superclass version in Node creates a new
+        // slave object, which we don't need.
+
+        final boolean held = form.getBoolean("held");
+        setHeld(held);
+
+        return this;
+    }
+
+    /**
+     * If true, hold onto the node after the associated build is completed.  Do not release it to NodePool to get
+     * cleaned up.
+     *
+     * @return true if node should be held.
+     */
+    public boolean isHeld() {
+        // check for optional "force hold" property, primarily useful for debugging.
+        final String forceHoldStr = System.getProperty(FORCE_HOLD_PROPERTY, "false");
+        final boolean forceHold = Boolean.valueOf(forceHoldStr);
+
+        return held || forceHold;
+    }
+
+    /**
+     * Set to true in order to hold the node
+     *
+     * @param held whether to hold the node
+     */
+    public void setHeld(boolean held) {
+        if (held) {
+            // tell NodePool to hold the node for us.
```

Comment:
`NodePoolNode.hold` ?

Reply:
I have the actual hold done when the build completes, in lieu of deletion.  In the following scenario, holding early could get awkward:

1. user checks "hold" checkbox
2. NodePoolNode.hold() - this sets the state to "hold" releases the lock on the node
3. user un-checks "hold" checkbox
4. We re-acquire the lock and change state to "in-use"?  Is this something the client should be able to do or is the responsibility of NodePool?  I'm not sure, and doing it at the end of the build is consistent with how Zuul executes holds.